exercises/practice/killer-sudoku-helper/.meta/Example.roc

```diff
@@ -0,0 +1,27 @@
+module [combinations]
+
+Combination : List U8
+
+combinations : { sum : U8, size : U8, exclude ? List U8 } -> List Combination
+combinations = \{ sum, size, exclude ? [] } ->
+    help = \target, digits ->
+        if target == 0 then
+            [[]]
+            else
+
+        when digits is
+            [] -> [[]]
+            [single] -> if single == target then [[single]] else []
+            [first, .. as rest] ->
+                if first > target then
+                    []
+                    else
+
+                help (target - first) rest
+                |> List.map \combi -> combi |> List.prepend first
```

Comment:
```suggestion
                |> List.map \combi -> combi |> List.append first
```

Reply:
Unfortunately if I use `List.append`, then the combinations end up in descending order, e.g., `[4, 2, 1]` instead of `[1, 2, 4]`, and the exercise instructions require them to be in ascending order. I guess I could use `List.append` and end with `List.reverse`? Since the combinations are really short, I thought it would probably be simpler to use `List.prepend`, but on the other hand, from a pedagogical point of view, it's probably better to show the approach that will work both for long and short lists. Wdyt?

Reply:
After some thought, I think it's best to use `List.append` and reverse the lists at the end. It doesn't make any performance difference in this exercise, but it certainly would if the lists were longer, and it's probably best to avoid using `List.prepend` whenever possible.

Reply:
Ah I see, make sense. Sounds good!